src/character.cpp

```diff
@@ -596,6 +596,54 @@ std::vector<item_location> Character::nearby( const std::function<bool(const ite
     return res;
 }
 
+long int Character::i_add_to_container(const item &it)
+{
+    long int charges = it.charges;
+    
+    if( it.is_ammo() && (!it.has_flag( "TEMP_UNLOADING" ) || (it.has_flag( "TEMP_UNLOADING" ) && charges == 1) ) ) {
```

Comment:
This whole `TEMP_UNLOADING` thing is confusing and needs a lot of comments.
What exactly is being checked in this line? Why exactly 1 charge?